db/model/aspect.js

```diff
@@ -318,17 +322,14 @@ module.exports = function aspect(seq, dataTypes) {
             }
           } else if (inst.isPublished) {
             const instChanged = {};
-            Object.keys(inst._changed).forEach((key) => {
-              instChanged[key] = inst[key];
-            });
+            Object.keys(inst._changed)
+            .filter(key => inst._changed[key])
+            .forEach(key => instChanged[key] = inst[key]);
             redisOps.hmSet(aspectType, inst.name, instChanged);
           }
         }
 
-        if (inst.changed('isPublished') &&
-          inst.previous('isPublished') &&
-          !inst.getDataValue('isPublished') ||
-          inst.changed('name')) {
+        if (isPublishedChanged && !inst.isPublished || nameChanged) {
```

Comment:
Generally I find lines like this easier to read and understand if they're written like this (explicit grouping), and we only need to evaluate the AND'ed expressions if the OR expression is true, so put that first?

`if (nameChanged || (isPublishedChanged && !inst.isPublished)) {`